ooniprobe/Model/NetworkMeasurement.mm

```diff
@@ -552,36 +365,52 @@ -(void)run{
 }
 
 -(void) run_test {
-    [super run_test];
-    setup_idempotent();
-    mk::nettests::HttpHeaderFieldManipulationTest()
-    .set_options("geoip_country_path", [geoip_country UTF8String])
-    .set_options("geoip_asn_path", [geoip_asn UTF8String])
-    .set_options("save_real_probe_ip", include_ip)
-    .set_options("save_real_probe_asn", include_asn)
-    .set_options("save_real_probe_cc", include_cc)
-    .set_options("no_collector", !upload_results)
-    .set_options("software_name", [@"ooniprobe-ios" UTF8String])
-    .set_options("software_version", [software_version UTF8String])
-    .set_output_filepath([[self getFileName:@"json"] UTF8String])
-    .set_error_filepath([[self getFileName:@"log"] UTF8String])
-    .set_verbosity(VERBOSITY)
-    .on_progress([self](double prog, const char *s) {
-        [self updateProgress:prog];
-    })
-    .on_log([self](uint32_t type, const char *s) {
-#ifdef DEBUG
-        NSLog(@"%s", s);
-#endif
-    })
-    .on_entry([self](std::string s) {
-        [self on_entry_hhfm:s.c_str()];
-    })
-    .start([self]() {
+    mk::nettests::HttpHeaderFieldManipulationTest test;
+    [super run_test:test];
+    test.on_entry([self](std::string s) {
+        [self on_entry:s.c_str()];
+    });
```

Comment:
It seems to me that `on_entry` is equal for all tests. Cannot you put this into the common code. If I understand OO correctly, calling `on_entry` in a parent class will resolve to the `on_entry` of the real class.

Reply:
Otherwise, you can perhaps pass the pointer to the method to the super class common-init method?

Reply:
The super class won't know that there is an on_entry in the bottom class